code/game/atom/atom_appearance.dm

```diff
@@ -56,6 +56,8 @@
 				continue
 			if(istext(maybe_not_an_atom) || isicon(maybe_not_an_atom))
 				continue
+			if(maybe_not_an_atom.layer < 0 && (maybe_not_an_atom.pixel_x || maybe_not_an_atom.pixel_y))
+				stack_trace("Float layer added to [src] ([type]) with pixel_x and pixel_y set on an overlay [maybe_not_an_atom.icon]/[maybe_not_an_atom.icon_state]")
```

Comment:
Actually going over our overlay usage, we would need to remove this - this is (sometimes) intentional, so stack tracing would be inappropriate.